cds_ils/migrator/utils.py

```diff
@@ -154,3 +166,130 @@ def clean_created_by_field(record):
         record["created_by"] = {"type": "script", "value": "migration"}
 
     return record
+
+
+def get_acq_ill_notes(record):
+    """Extract notes for acquisition and ILL records."""
+    # NOTE: library notes are usually empty dict stringified '{}'
+    result = ""
+    cost = record.get("cost")
+    if cost:
+        result += "cost: {0}\n\n".format(cost)
+
+    item_info = record.get("item_info")
+    if item_info:
+        result = "item info: {0}\n\n".format(item_info.strip())
+
+    comments = record.get("borrower_comments")
+    if comments:
+        result = "borrower comments: {0}\n\n".format(comments.strip())
+
+    notes = record.get("library_notes")
+    if not notes:
+        return result
+
+    try:
+        if json.loads(notes):
+            result += "library notes: {0}\n".format(notes.strip())
+    except json.decoder.JSONDecodeError:
+        # At this point if we failed to parse json, we have notes as string
+        result += "library notes: {0}\n".format(notes.strip())
+
+    return result
+
+
+def get_cost(record):
+    """Parse CDS cost string to extract currency and value."""
+    SUPPORTED_CURRENCIES = ["EUR", "USD", "GBP", "CHF"]
+    # NOTE: We will try to extract and migrate only the most common case, which
+    # comes to the form "EUR 88.40". If it comes in any other form will
+    # store it in notes.
+
+    try:
+        [currency, value] = record.get("cost").split(" ")
+        if currency.upper() in SUPPORTED_CURRENCIES:
+            return {"currency": currency.upper(), "value": float(value)}
+    except Exception:
+        # We don't care, we will check again at get_acq_ill_notes()
+        pass
+
+
+def get_patron_pid(record):
+    """Get patron_pid from existing users."""
+    user = get_user_by_legacy_id(record["id_crcBORROWER"])
+    if not user:
+        # user was deleted, fallback to the AnonymousUser
+        anonym = current_app.config["ILS_PATRON_ANONYMOUS_CLASS"](-2).dumps()
+        patron_pid = anonym["id"]
+    else:
+        patron_pid = user.pid
+    return str(patron_pid)
+
+
+MIGRATION_DOCUMENT_PID = "DOCPID-44444"
+MIGRATION_LIBRARY_PID = "LIBPID-44444"
+MIGRATION_VENDOR_PID = "VENPID-44444"
+
+def get_migration_document_pid():
+    """Get the PID of the dedicated migration document."""
+    return current_app_ils.document_record_cls.get_record_by_pid(
+        MIGRATION_DOCUMENT_PID
+    ).pid.pid_value
+
+
+def create_document():
+    data = {
+        "pid": MIGRATION_DOCUMENT_PID,
+        "title": "Migrated Unknown Document",
+        "authors": [{"full_name": "Legacy CDS service"}],
+        "publication_year": "2020",
+        "document_type": "BOOK",
+        "restricted": True,
+        "notes": """This document is used whenever we document information
+        is requierd and not provided, in order to migrate data from CDS.
```

Comment:
Typos?

`whenever we document information is required...` ?

```suggestion
        is required and not provided, in order to migrate data from CDS.
```